iocore/hostdb/HostDB.cc

```diff
@@ -1401,6 +1401,9 @@ HostDBContinuation::remove_and_trigger_pending_dns()
   }
   EThread *thread = this_ethread();
   while ((c = qq.dequeue())) {
+    if (c->action.cancelled) {
```

Comment:
Checking action state looks reasonable. But this make me wondering it should be also checked in the above while loop (line 1392).

Reply:
I felt safer checking at the point of calling the action rather than earlier in the process. Maybe processing earlier actions can cancel later ones? But hopefully that is not the case. I'll pull it earlier and we can see how that impacts things.

Thanks for the review.